js/FileManager.js

```diff
@@ -0,0 +1,156 @@
+var InsightMakerFileExtension = ".InsightMaker";
+
+// Append file extension to file (if not already there)
+function append_file_extension(filename,extension) {
+	var extension_position=filename.length-extension.length;
+	var current_extension=filename.substring(extension_position,filename.length);
+	if(current_extension.toLowerCase()!=extension.toLowerCase()) {
+		filename+=extension;
+	}
+	return filename;
+}
+
+// Set the title to include the model name
+function set_title(filename) {
+	var title;
+	if(filename!=null) {
```

Comment:
I prefer explicitly stating !=null, maybe because if (filename) could mean it's not undefined OR it's not null. BUT I will make it if(filename) anyway, beause I think most other programmers prefer your way of writting it :)